examples/lock-app/k32w/main/AppTask.cpp

```diff
@@ -469,6 +485,37 @@ void AppTask::BleHandler(AppEvent * aEvent)
     }
 }
 
+#ifdef CONFIG_CHIP_NFC_COMMISSIONING
+void AppTask::ThreadProvisioningHandler(const ChipDeviceEvent * event, intptr_t)
+{
+    if (event->Type == DeviceEventType::kCHIPoBLEAdvertisingChange && event->CHIPoBLEAdvertisingChange.Result == kActivity_Stopped)
+    {
+        if (!NFCMgr().IsTagEmulationStarted())
+        {
+            K32W_LOG("NFC Tag emulation is already stopped!");
+        }
+        else
+        {
+            NFCMgr().StopTagEmulation();
+            K32W_LOG("Stopped NFC Tag Emulation!");
+        }
+    }
+    else if (event->Type == DeviceEventType::kCHIPoBLEAdvertisingChange &&
+             event->CHIPoBLEAdvertisingChange.Result == kActivity_Started)
+    {
+        if (NFCMgr().IsTagEmulationStarted())
+        {
+            K32W_LOG("NFC Tag emulation is already started!");
+        }
+        else
+        {
+            ShareQRCodeOverNFC(chip::RendezvousInformationFlags(chip::RendezvousInformationFlag::kBLE));
```

Comment:
According to the specification: `The device SHALL require explicit trigger of NFC pairing mode via a physical interaction on the device.`. Is this the case for K32W? What happens if someone configures the BLE advertisement on boot.

Reply:
@LuDuda , agree that the spec states that NFC pairing mode should be started via physical interaction (push of a button). However, I can't find anything related to BLE advertisement auto-start in the specification - could you point me out the spec section?

I see the following define in the implementation:

```
/**
 * CHIP_DEVICE_CONFIG_CHIPOBLE_ENABLE_ADVERTISING_AUTOSTART
 *
 * Enable CHIPoBLE advertising start automatically after device power-up.
 *
 * CHIP's device may start advertising automatically only if its all primary device
 * functions are within a CHIP network. Device providing unrelated to CHIP functionalities
 * should not start advertising automatically after power-up.
 *
 * TODO: Default value should be changed to 0, after all platforms will implement enabling
 *       advertisement in their own way.
 */
#ifndef CHIP_DEVICE_CONFIG_CHIPOBLE_ENABLE_ADVERTISING_AUTOSTART
#define CHIP_DEVICE_CONFIG_CHIPOBLE_ENABLE_ADVERTISING_AUTOSTART 1
#endif
```

So my understanding is that in the end this define should be put to 0.

Reply:
My comment was more about the relation between NFC and BLE advertisement in the code. More specifically if someone enables BLE adv on boot, NFC will automatically start which sort of violates current spec (and spec allows to advertise without user physical interaction). Anyway, I'm approving.